docs/compilation/performance_tips.md

```diff
@@ -0,0 +1,645 @@
+
+    return result
+```
+
+There are two table lookups within the loop body, one for `>>` and one for `%`.
+
+This implementation is not optimal though, since the same output can be achieved with just a single table lookup:
+
+```python
+def add_bitmaps(x, y):
+    result = fhe.zeros((N,))
+    carry = 0
+
+    addition = x + y
+    for i in range(N):
+        addition_and_carry = addition[i] + carry
+        carry = addition_and_carry >> 1
+        result[i] = addition_and_carry - (carry * 2)
+
+    return result
+```
+
+It was possible to do this because the original operations had a mathematical equivalence with the optimized operations and optimized operations achieved the same output with less table lookups!
+
+Here is the full code example and some numbers for this optimization:
+
+```python
+import numpy as np
+from concrete import fhe
+
+N = 32
+
+def add_bitmaps_naive(x, y):
+    result = fhe.zeros((N,))
+    carry = 0
+
+    addition = x + y
+    for i in range(N):
+        addition_and_carry = addition[i] + carry
+        carry = addition_and_carry >= 2
+        result[i] = addition_and_carry % 2
+
+    return result
+
+def add_bitmaps_optimized(x, y):
+    result = fhe.zeros((N,))
+    carry = 0
+
+    addition = x + y
+    for i in range(N):
+        addition_and_carry = addition[i] + carry
+        carry = addition_and_carry >> 1
+        result[i] = addition_and_carry - (carry * 2)
+
+    return result
+
+inputset = fhe.inputset(fhe.tensor[fhe.uint1, N], fhe.tensor[fhe.uint1, N])
+for (name, implementation) in [("naive", add_bitmaps_naive), ("optimized", add_bitmaps_optimized)]:
+    compiler = fhe.Compiler(implementation, {"x": "encrypted", "y": "encrypted"})
+    circuit = compiler.compile(inputset)
+
+    print(
+        f"{name:>9} implementation "
+        f"-> {int(circuit.programmable_bootstrap_count)} table lookups "
+        f"-> {int(circuit.complexity):_} complexity"
+    )
+```
+
+prints:
+
+```
+    naive implementation -> 63 table lookups -> 2_427_170_697 complexity
+optimized implementation -> 32 table lookups -> 1_224_206_208 complexity
+```
+
+which is almost half the amount of table lookups and ~2x less complexity for the same operation!
+
+### By changing the implementation strategy of complex operations
+
+Concrete provides multiple implementation strategies for these complex operations:
+
+- [comparisons (<,<=,==,!=,>=,>)](../core-features/comparisons.md)
+- [bitwise operations (<<,&,|,^,>>)](../core-features/bitwise.md)
+- [minimum and maximum operations](../core-features/minmax.md)
+- [multivariate extension](../core-features/extensions.md#fhemultivariatefunction)
+
+{% hint style="info" %}
+The default strategy is the one that doesn't increase the input bit width, even if it's less optimal than the others. If you don't care about the input bit widths (e.g., if the inputs are only used in this operation), you should definitely change the default strategy.
+{% endhint %}
```

Comment:
this part for me is complicated: I would just say what you have after, ie 

> Choosing the correct strategy can lead to big speedups. So if you are not sure which one to use, you can compile with different strategies and compare the complexity.

Reply:
I wrote this part to answer what the default strategy is. Without this info, users could wonder why `XYZ` is not the default strategy.